### PR TITLE
Added pre-commit hooks and updated readme.

### PR DESCRIPTION
**Pull request by ryanbnl**
Pre-commit hooks, need to be tested (on a different machine so this is a draft for now)

### CODE REVIEW

**Review by dslmeinte — CHANGES_REQUESTED** [review]

**Comment by dslmeinte** [comment]
By the looks of it, this PR mostly needs to tested out by other people than the PR creator.

File: hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# If there are whitespace errors, print the offending file names and fail.
+exec git diff-index --check --cached $against --
+
+# Check that all of the schema files have been formatted correctly
+make check-formatting
+
+# Check all examples
+make validate-examples
+
+# Check test cases which must be catogorized as valid
```
Comment (dslmeinte) [inline hooks/pre-commit:57]: Typo: "categorized"

File: README.md [hunk 2]
```diff
@@ -77,3 +78,13 @@ To validate all of the examples:
...
 	make validate-invalid-tests
+
+
+## Committing: Git hooks
+
+This project uses Git hooks to validate the JSON files and the above test cases.
+To use the hooks, configure Git to use the hooks we provide, as follows:
+
+```
+git config core.hooksPath hooks
```
[the remark this answers is not in the document]
Comment (dslmeinte) [reply README.md:89]: (self-answer:) We could, but then you wouldn't be able to commit that file.

**Comment by ryanbnl** [comment]
Tested with your changes and it's OK.
Because we've both committed we will need to merge ignoring the reviews - but we have both reviewed and agreed so have still followed the process.

**Pull request merged**